src/Analyzer.Utilities/SyntaxGeneratorExtensions.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Analyzer.Utilities
+{
+    public static class SyntaxGeneratorExtensions
```

Comment:
@srivatsn This is a completely different `SyntaxGeneratorExtensions` class than the one you commented on in the initial commit. This one includes extension methods to DRY out the creation of declarations for comparison operators, `object.Equal(object obj)`, and `object.GetHashCode()`, which occurred several times throughout the code.